File: newrust/src/mixers.rs
/*
   Hackflight mixuer support

   Copyright (C) 2022 Simon D. Levy

   MIT License
 */

use crate::Demands;
use crate::Motors;

/*
   Mixer values for quad-X frames using Betaflight motor layout:

   4cw   2ccw
   \ /
   ^
   / \
   3ccw  1cw

   Copyright (C) 2022 Simon D. Levy

   MIT License
 */
pub fn quad_xbf(demands:Demands) -> Motors {

    Motors {

        // right rear
        m1: demands.throttle - demands.roll + demands.pitch + demands.yaw, 

        // right front
        m2: demands.throttle - demands.roll - demands.pitch - demands.yaw, 

        // left rear
        m3: demands.throttle + demands.roll + demands.pitch - demands.yaw, 

        // left front
        m4: demands.throttle + demands.roll - demands.pitch + demands.yaw  
    }
}

File: attic/webots/controllers/playback-demands/mixers.py
def runcf(demands):
    '''
    Motor mixer for Crazyflie motor layout
    '''

    t, r, p, y = demands

    m1 = t - r - p  + y  # Front right
    m2 = t - r + p  - y  # Back right
    m3 = t + r + p  + y  # Back left
    m4 = t + r - p  - y  # Front left

    return m1, m2, m3, m4
 

